air/src/stack/op_flags/tests.rs

```diff
@@ -0,0 +1,123 @@
+use crate::stack::op_flags::get_op_index;
+
+use super::{
+    generate_evaluation_frame, OpFlags, DEGREE_4_OPCODE_ENDS, DEGREE_4_OPCODE_STARTS,
+    DEGREE_6_OPCODE_ENDS, DEGREE_6_OPCODE_STARTS, DEGREE_7_OPCODE_ENDS, DEGREE_7_OPCODE_STARTS,
+    NUM_DEGREE_4_OPS, NUM_DEGREE_6_OPS, NUM_DEGREE_7_OPS,
+};
+use vm_core::{ONE, ZERO};
+
+#[test]
+/// Asserts the op flag to ONE for degree 7 operation which is being executed in the current
+/// frame; assert all the other operation flags to ZERO as they are not present in the current
+/// trace.
+fn degree_7_op_flags() {
+    for i in DEGREE_7_OPCODE_STARTS..=DEGREE_7_OPCODE_ENDS {
+        // frame initialised with a degree 7 operation using it's unique opcode.
+        let frame = generate_evaluation_frame(i);
+
+        // All the operation flags are generated for the given frame.
+        let op_flags = OpFlags::new(frame);
+
+        // index of the operation flag in the op_flag's degree seven array.
+        let idx_in_degree7_flags = get_op_index(i as u8);
+
+        // Asserts operation flag of degree 7 operation being executed to ONE.
+        assert_eq!(op_flags.degree7_op_flags[idx_in_degree7_flags], ONE);
+
+        // Asserts operation flags not present in the trace to ZERO.
+        for i in 0..NUM_DEGREE_7_OPS {
+            if i != idx_in_degree7_flags {
+                assert_eq!(op_flags.degree7_op_flags[i], ZERO)
+            }
+        }
+
+        // Asserts all degree 6 operation flags to ZERO as the input operation is of degree 7.
+        for i in 0..NUM_DEGREE_6_OPS {
+            assert_eq!(op_flags.degree6_op_flags[i], ZERO)
+        }
+
+        // Asserts all degree 4 operation flags to ZERO as the input operation is of degree 7.
+        for i in 0..NUM_DEGREE_4_OPS {
+            assert_eq!(op_flags.degree4_op_flags[i], ZERO)
+        }
+    }
+}
+
+#[test]
+/// Asserts the op flag to ONE for degree 6 operation which is being executed in the current
+/// frame; assert all the other operation flags to ZERO as they are not present in the current
+/// trace.
```

Comment:
very minor nit: we usually place attributes after comments.

Reply:
thanks, I've fixed it in the commit [656e6d7](https://github.com/maticnetwork/miden/pull/380/commits/656e6d713af08e3f10b562d96860b2ef0c4c07dc).